src/navier_stokes/Tnavier_stokes_elements.h

```diff
@@ -745,6 +745,11 @@ namespace oomph
                              public virtual ElementWithZ2ErrorEstimator
 
   {
+  public:
+    // publically expose template parameters
+    static const unsigned _DIM_ = DIM;
+    static const unsigned _NNODE_1D_ = 3;
```

Comment:
Think we've discussed this one before - it's said to be a **big no-no**:
> Reserved in any scope, including for use as implementation macros:
> - identifiers beginning with an underscore followed immediately by an uppercase letter
> - identifiers containing adjacent underscores (or "double underscore")

See, e.g., discussion [here](https://stackoverflow.com/a/228797).

Names following this convention are *reserved* so they might not break your code right now but they have the possibility to in the future.

How about `OOMPH_DIM` and `OOMPH_NNODE_1D` if you want to keep it "oomph"-generic (rather than element-specific)?

Reply:
updated the naming convention to TEMPLATE_PARAMETER_ as a prefix, and dropped the trailing underscore